generator/src/main/java/dev/cdevents/generator/PreprocessSchemas.java

```diff
@@ -0,0 +1,66 @@
+package dev.cdevents.generator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+public class PreprocessSchemas {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final Logger log = LoggerFactory.getLogger(PreprocessSchemas.class);
+
+    public static void main(String[] args) throws IOException {
+        if (args == null || args.length != 1) {
+            System.err.println("Usage: PreprocessSchemas <schema_directory_path>");
+            throw new IllegalArgumentException("Schema directory path arguments not passed to PreprocessSchemas");
+        }
+        String parentBaseDir = args[0];
+        String specSchemaDir = parentBaseDir + File.separator + "spec" + File.separator + "schemas";

```

Comment:
Why is the parent base passed in as an argument, but everything else is an expected format, e.g. the `<parent>/spec/schemas`? Is this for testing? I wonder if it just makes sense to pass in the full `schemas` path instead

Reply:
Updated to pass `spec.schemas.dir` as an argument.